controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -141,3 +141,72 @@ func (a *Adapter) getDeploymentTargetForEnvironment(environment *applicationapiv
 
 	return deploymentTarget, nil
 }
+
+// EnsureEphemeralEnvironmentsCleanedUp will ensure that ephemeral environment(s) are cleaned up.
+func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (reconciler.OperationResult, error) {
+	testEnvironment := a.environment
+
+	isEphemeral := false
+	for _, tag := range testEnvironment.Spec.Tags {
+		if tag == "ephemeral" {
+			isEphemeral = true
+			break
+		}
+	}
+
+	if isEphemeral {
+		dtc, err := a.loader.GetDeploymentTargetClaimForEnvironment(a.client, a.context, testEnvironment)
+		if err != nil || dtc == nil {
+			a.logger.Error(err, "Failed to find deploymentTargetClaim defined in environment", "environment.Name", testEnvironment.Name)
+			return reconciler.RequeueWithError(err)
+		}
+
+		dt, err := a.loader.GetDeploymentTargetForDeploymentTargetClaim(a.client, a.context, dtc)
+		if err != nil || dt == nil {
+			a.logger.Error(err, "Failed to find deploymentTarget defined in deploymentTargetClaim", "deploymentTargetClaim.Name", dtc.Name)
+			return reconciler.RequeueWithError(err)
+		}
+
+		binding, err := a.loader.FindExistingSnapshotEnvironmentBinding(a.client, a.context, a.application, testEnvironment)
+		if err != nil || binding == nil {
+			a.logger.Error(err, "Failed to find snapshotEnvironmentBinding associated with environment", "environment.Name", testEnvironment.Name)
+			return reconciler.RequeueWithError(err)
+		}
+
+		a.logger.Info("Deleting deploymentTarget", "deploymentTarget.Name", dt.Name)
+		err = a.client.Delete(a.context, dt)
+		if err != nil {
+			a.logger.Error(err, "Failed to delete the deploymentTarget")
```

Comment:
Suggestion:
We can include `"deploymentTarget.Name", dt.Name` in error msg also, so its easier to debug.
Same applies for rest of the error msgs.